supercollider/server.py

```diff
@@ -13,188 +16,131 @@ def __init__(self, hostname="127.0.0.1", port=57110):
         Create a new Server object, which is a local representation of a remote
         SuperCollider server.
 
+        Supercollider communication is (unfortunatelly for UDP) made in the same
+        port used by the client. Hence, the OSC server and the UDP client should
+        share the same port. Setting this up is possible, but slightly harder.
+        Check this github issue to see how this is possible with pythonosc:
+        https://github.com/attwad/python-osc/issues/41
+
         Args:
             hostname (str): Hostname or IP address of the server
             port (int): Port of the server
         """
-        self.client_address = liblo.Address(hostname, port)
-
-        #-----------------------------------------------------------------------
-        # Set up OSC server and default handlers for receiving messages.
-        #-----------------------------------------------------------------------
-        self.osc_server = liblo.Server(random.randint(57200, 57900))
-        self.osc_server.add_method(None, None, self._osc_handler)
-        self.osc_server_thread = threading.Thread(target=self._osc_server_read)
-        self.osc_server_thread.setDaemon(True)
+
+        # UDP Client for sending messages
+        self.client_address = (hostname, port)
+        self.sc_client = SimpleUDPClient(hostname, port)
+        self.sc_client._sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.sc_client._sock.bind(('',0))
+
+        # OSC Server for receiving messages.
+        self.dispatcher = Dispatcher()
+        self.osc_server_address = ("127.0.0.1", self.sc_client._sock.getsockname()[1])
+        ThreadingOSCUDPServer.allow_reuse_address = True
+        self.osc_server = ThreadingOSCUDPServer(self.osc_server_address, self.dispatcher)
+
+        self.osc_server_thread = Thread(target=self._osc_server_listen, daemon=True)
         self.osc_server_thread.start()
-        self.handlers = {
-            "/n_set": {},
-            "/b_info": {},
-            "/b_setn" : {},
-            "/done": {},
-            "/status.reply": None,
-            "/version.reply": None,
-            "/synced": None,
-            "/g_queryTree.reply": None
-        }
-
-        #-----------------------------------------------------------------------
-        # Necessary when querying a node ID for add actions.
-        #-----------------------------------------------------------------------
+
+        # Callbacks witout await_response timeout
+        # self.dispatcher.map("/example", self.dummy_handler)
```

Comment:
Commented-out code, should be deleted

Reply:
Solved in [9bd584f](https://github.com/ideoforms/python-supercollider/pull/12/commits/9bd584f53a5ec7a0eed072ca67869c326910dcdc)